File: scripts/build_standalone_mvp.mjs
#!/usr/bin/env node

import { copyFileSync, existsSync, readFileSync, writeFileSync } from "node:fs";
import { resolve } from "node:path";

const root = resolve(import.meta.dirname, "..");
const mvpDir = resolve(root, "website/mvp");
const outputPath = resolve(mvpDir, "wm-2026-weather-fit-mvp.html");
const staticAssetNames = [
  "apple-touch-icon.png",
  "favicon-16.png",
  "favicon-32.png",
  "icon-192.png",
  "icon-512.png",
  "icon-1024.png",
  "og-image.png",
  "site.webmanifest",
];

function readMvpFile(name) {
  return readFileSync(resolve(mvpDir, name), "utf8");
}

function inlineScript(source) {
  return source.replaceAll("</script", "<\\/script");
}

let html = readMvpFile("index.html");
const css = readMvpFile("styles.css");
const data = readMvpFile("data.js");
const mapData = readMvpFile("map-data.js");
const app = readMvpFile("app.js");
const outputDir = resolve(outputPath, "..");

html = html
  .replace('<link rel="stylesheet" href="./styles.css">', `<style>\n${css}\n</style>`)
  .replace('<script src="./data.js"></script>', `<script>\n${inlineScript(data)}\n</script>`)
  .replace('<script src="./map-data.js"></script>', `<script>\n${inlineScript(mapData)}\n</script>`)
  .replace('<script src="./app.js"></script>', `<script>\n${inlineScript(app)}\n</script>`);

writeFileSync(outputPath, html, "utf8");
for (const name of staticAssetNames) {
  const sourcePath = resolve(mvpDir, name);
  const targetPath = resolve(outputDir, name);
  if (existsSync(sourcePath) && sourcePath !== targetPath) {
    copyFileSync(sourcePath, targetPath);
  }
}
console.log(outputPath);

File: python/pipelines/build_standalone_mvp.py
"""Build one self-contained HTML file for the static website MVP."""

from __future__ import annotations

import argparse
from pathlib import Path

ROOT_DIR = Path(__file__).resolve().parents[2]
MVP_DIR = ROOT_DIR / "website" / "mvp"


def _read(name: str) -> str:
    return (MVP_DIR / name).read_text(encoding="utf-8")


def _inline_script(source: str) -> str:
    return source.replace("</script", "<\\/script")


def build_standalone_mvp(output: str = "website/mvp/wm-2026-weather-fit-mvp.html") -> dict[str, str | int]:
    """Inline CSS, data and scripts into a single browser-openable HTML file."""

    html = _read("index.html")
    html = html.replace('<link rel="stylesheet" href="./styles.css">', f"<style>\n{_read('styles.css')}\n</style>")
    html = html.replace('<script src="./data.js"></script>', f"<script>\n{_inline_script(_read('data.js'))}\n</script>")
    html = html.replace('<script src="./map-data.js"></script>', f"<script>\n{_inline_script(_read('map-data.js'))}\n</script>")
    html = html.replace('<script src="./experience.js"></script>', f"<script>\n{_inline_script(_read('experience.js'))}\n</script>")
    html = html.replace('<script src="./app.js"></script>', f"<script>\n{_inline_script(_read('app.js'))}\n</script>")

    output_path = ROOT_DIR / output
    output_path.parent.mkdir(parents=True, exist_ok=True)
    output_path.write_text(html, encoding="utf-8")
    return {"output": str(output_path), "bytes": output_path.stat().st_size}


def main(argv: list[str] | None = None) -> int:
    parser = argparse.ArgumentParser(description="Build standalone website MVP HTML")
    parser.add_argument("--output", default="website/mvp/wm-2026-weather-fit-mvp.html")
    args = parser.parse_args(argv)
    print(build_standalone_mvp(args.output))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
